Functions/Initialization/xASL_init_InitializeMutex.m

```diff
@@ -63,17 +63,99 @@
 	fprintf(2,['ERROR in module_' x.ModuleName ': mutex is locked: %s in %s\n'], x.settings.MUTEXID, x.dir.LockDir);
-	fprintf('This means that this module is currently being parallel processed by another Matlab instance/worker\n');
+	fprintf('\n');
+    fprintf('This means that this module is currently being parallel processed by another Matlab instance/worker\n');
 	fprintf('If this is not the case, the locked folder needs to be removed before proceeding\n');
 	fprintf('Also, check that there is no filesystem permission issue\n');
     fprintf('Otherwise this error can be ignored\n');
-	return;
-end
+    fprintf('\n');
+
+    x.mutex.bAnyModuleLocked = true;
+else
+    
+    %% 4) Check if any module for this subject is locked
+    % Here we check if any other module for the same subject is locked.
+    % So whereas xASL_GoNoGo in step 2 above only checks the current
+    % module, here we check if ANY module for this subject is locked.
+
+    if ~strcmpi(x.ModuleName, 'population')
+
+        % If the current module is the population module, we don't need to
+        % perform this check because the population module is cannot be
+        % iterated over subjects, it is always run once.
+        % Plus, at the start of the population module itself, we already
+        % check if nWorkers>1, in which case we skip the population module
+        % (== the population module should not run in parallel).
+        if ~isfield(x, 'mutex') || ~isprop(x.mutex, 'Root')
+            warning('mutex field missing, something odd going wrong');
+        else
+            [~, subjectName] = fileparts(x.mutex.Root);
+            lockDir = fullfile(x.dir.xASLDerivatives, 'lock');
+						
+			% Go through all module that have lock dirs already and are thus potentially locked
+			checkModuleList = xASL_adm_GetFileList(lockDir, '.*', 'FPList', [], 1);
+            for iCheckModule=1:length(checkModuleList)
+                [~, checkModuleName] = fileparts(checkModuleList{iCheckModule});
 
+				% Verify that it is really a directory with an ASL module lock files
+                if strcmp(checkModuleName(1:12), 'xASL_module_')
+                    checkModuleName = checkModuleName(13:end); % Extract the name of the module
+                    if strcmpi(checkModuleName, x.ModuleName)
+                        % this is the current module, so we can continue if this is locked
+                    elseif strcmpi(checkModuleName, 'population') || strcmpi(checkModuleName, 'import')
+                        % for population or import modules, we can continue if other modules are locked
+                    else
+                        % For BIDS2Legacy, no visit suffix is given. For other modules, a suffix is given
+						bLockedFolders = false;
+						
+						if strcmp(x.ModuleName, 'BIDS2Legacy') && ~strcmp(checkModuleName, 'BIDS2Legacy')
+							% In case we are in BIDS2Legacy and are checking other modules, we need to add a suffix

```

Comment:
In this case (we are in a BIDS2Legacy session but checking for possibly running structural module), we actually need to check all visits, so we need to go through all visits.
I have fixed that.
HOWEVER, this will be even more complicated for the ASL module. Because we have to check the lock folders inside the ASL sessions :) So we need to go through all visits, and for ASL also for all sessions. I haven't programmed that yet, but you or I need to do that. I can do that tomorrow.

Reply:
That's easy right, by creating a recursive list of the lock folders?

Reply:
As discussed. We don't want to load the entire directory tree, but only check at the specific folders. I have fixed that for structural and I will also (specifically) load the ASL directories for all sessions. Yes, easy to do, I just didn't have the extra 10 minutes to do that properly, so I only wrote it to a comment :)